modules/vault/options.go

```diff
@@ -0,0 +1,82 @@
+package vault
+
+import "fmt"
+
+// Option is a function type for modifying a Vault configuration
+type Option func(*Config)
+
+// Config is a struct that contains various configuration options for the Vault
+type Config struct {
+	imageName    string              // The name of the Docker image to use for the Vault
+	token        string              // The root token to use for the Vault
+	port         int                 // The port number to use for the Vault
+	secrets      map[string][]string // A map of secret paths to their respective secret values
+	initCommands []string            // A list of commands to execute on Vault initialization
```

Comment:
I think this was based on the Java implementation but TBH it was confusing some times, see [this comment](https://github.com/testcontainers/testcontainers-java/issues/5510#issuecomment-1179451354). IMO, I would rely more on [executing a command](https://golang.testcontainers.org/features/override_container_command/)  or [copying a script](https://golang.testcontainers.org/features/copy_file/) and executing it later. WDYT?

Reply:
good idea but `Cmd` is running before `WaitingFor` so it try to execute a command before container running. Is there any way to execute command after `WaitingFor`? If there is not, copying a script is more sense.

Reply:
I solved it using `WaitingFor`. Can you review it again?